lib/vector-core/src/metrics/mod.rs

```diff
@@ -63,6 +63,18 @@ fn init(recorder: VectorRecorder) -> Result<()> {
         return Ok(());
     }
 
+    /*
+    // If we've already installed our recorder, make this a no-op, _unless_ the installed recorder is actually a
+    // different type.
+    if let Some(controller) = CONTROLLER.get() {
+        match (&controller.recorder, recorder) {
+            (VectorRecorder::Global(_), VectorRecorder::ThreadLocal) |
+            (VectorRecorder::ThreadLocal, VectorRecorder::Global(_)) => return Err(Error::AlreadyInitialized),
+            _ => return Ok(()),
+        }
+    }
+    */
```

Comment:
No, it was not.

I was totally missing where you added per-thread metrics support and so I thought I needed to write this myself.. then realized my mistake, but commented it out instead of deleting it.

I've resolved this by removing it in fbe2273.